test/test_device_adjacent_difference.cu

```diff
@@ -491,17 +502,24 @@ void Test(std::size_t elements)
 }
 
 
-void Test(std::size_t elements)
+template <typename NumItemsT>
+void Test(NumItemsT elements)
 {
-  Test<std::int32_t>(elements);
-  Test<std::uint32_t>(elements);
-  Test<std::uint64_t>(elements);
+  Test<std::int32_t, NumItemsT>(elements);
+  Test<std::uint32_t, NumItemsT>(elements);
+  Test<std::uint64_t, NumItemsT>(elements);
 }
 
 
-template <typename ValueT>
-void TestFancyIterators(std::size_t elements)
+template <typename ValueT,
+          typename NumItemsT>
+void TestFancyIterators(NumItemsT elements)
 {
+  if (elements == 0)
```

Comment:
Is this needed? We should test that empty inputs are well behaved.

Reply:
It's needed because this particular test doesn't work in case of `elements == 0` (uses `elements - 1`). Empty inputs are tested in `Test` and `TestCopy`. Iterators doesn't affect the underlying logic, so it's safe to skip here.